README.md

```diff
@@ -1,2 +1,27 @@
-# crypto
-Cryptographic primitives used in Polygon Miden rollup
+# Miden Crypto
+This crate contains cryptographic primitives used in Polygon Miden.
+
+## Hash
+[Hash module](./src/hash) provides a set of cryptographic hash functions which are used by Miden VM and Miden Rollup. Currently, these functions are:
+
+* [BLAKE3](https://github.com/BLAKE3-team/BLAKE3) hash function with 256-bit, 192-bit, or 160-bit output. The 192-bit and 160-bit outputs are obtained by truncating the 256-bit output of the standard BLAKE3.
+* [RPO](https://eprint.iacr.org/2022/1577) hash function with 256-bit output. This hash function is an algebraic hash function suitable for recursive STARKs.
```

Comment:
Should we mention what this function is called? `Rescue Prime Optimized`?

Reply:
I think it is fine to skip the full name here.